src/python/pants/backend/native/subsystems/native_build_step_settings.py

```diff
@@ -37,32 +37,42 @@ def get_compiler_option_sets_enabled_default_value(cls):
     return {"fatal_warnings": ["-Werror"]}
 
 
-class CCompileSettings(Subsystem):
-  options_scope = 'c-compile-settings'
+class CompileSettingsBase(Subsystem):
 
   @classmethod
   def subsystem_dependencies(cls):
-    return super(CCompileSettings, cls).subsystem_dependencies() + (
+    return super(CompileSettingsBase, cls).subsystem_dependencies() + (
       NativeBuildStepSettings.scoped(cls),
     )
 
+  header_file_extensions_default = None
```

Comment:
Out of curiousity - you wrote `@classproperty` - what criteria do you useto select when you'll use that vs this style?

Reply:
This was cherry-picked from a very old commit -- I think `@classproperty` with a `NotImplementedError` (perhaps this could be its own decorator?) is appropriate in probably all cases for public class fields now.

Reply:
Changed as per above!